tests/functional/iceberg/test_incremental_models.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+
+from pathlib import Path
+
+
+from dbt.tests.util import run_dbt, run_dbt_and_capture, rm_file, write_file
+
+_SEED_INCREMENTAL_STRATEGIES = """
+world_id,world_name,boss
+1,Yoshi's Island,Iggy
+2,Donut Plains,Morton
+3,Vanilla Dome,Lemmy
+4,Cookie Mountain,Temmy
+5,Forest of Illusion,Roy
+""".strip()
+
+_MODEL_BASIC_TABLE_MODEL = """
+{{
+  config(
+    materialized = "table",
+  )
+}}
+select * from {{ ref('seed') }}
+"""
+
+_MODEL_INCREMENTAL_ICEBERG_BASE = """
+{{{{
+  config(
+    materialized='incremental',
+    table_format='iceberg',
+    incremental_strategy='{strategy}',
+    unique_key="world_id",
+    external_volume = "s3_iceberg_snow",
+  )
+}}}}
+select * from {{{{ ref('upstream_table') }}}}
+
+{{% if is_incremental() %}}
+where world_id > 2
+{{% endif %}}
+"""
+
+_MODEL_INCREMENTAL_ICEBERG_APPEND = _MODEL_INCREMENTAL_ICEBERG_BASE.format(strategy="append")
+_MODEL_INCREMENTAL_ICEBERG_MERGE = _MODEL_INCREMENTAL_ICEBERG_BASE.format(strategy="merge")
+_MODEL_INCREMENTAL_ICEBERG_DELETE_INSERT = _MODEL_INCREMENTAL_ICEBERG_BASE.format(
+    strategy="delete+insert"
+)
+
+
+_QUERY_UPDATE_UPSTREAM_TABLE = """
+UPDATE {database}.{schema}.upstream_table set world_name = 'Twin Bridges', boss = 'Ludwig' where world_id = 4;
+"""
+
+_QUERY_UPDATE_UPSTREAM_TABLE_NO_EFFECT = """
+UPDATE {database}.{schema}.upstream_table set world_name = 'Doughnut Plains' where world_id = 2;
+"""
+
+
+class TestIcebergIncrementalStrategies:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"flags": {"enable_iceberg_materializations": True}}
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {
+            "seed.csv": _SEED_INCREMENTAL_STRATEGIES,
+        }
+
+    @pytest.fixture(scope="function", autouse=True)
+    def setup_class(self, project):
+        run_dbt(["seed"])
+        yield
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "upstream_table.sql": _MODEL_BASIC_TABLE_MODEL,
+            "append.sql": _MODEL_INCREMENTAL_ICEBERG_APPEND,
+            "merge.sql": _MODEL_INCREMENTAL_ICEBERG_MERGE,
+            "delete_insert.sql": _MODEL_INCREMENTAL_ICEBERG_DELETE_INSERT,
+        }
+
+    def test_incremental_strategies_build(self, project, setup_class):
+        run_results = run_dbt()
+        assert len(run_results) == 4
+
+    def __check_correct_operations(self, model_name, /, rows_affected, status="SUCCESS"):
+        run_results, stdout = run_dbt_and_capture(
```

Comment:
I would update this to `run_results, _`, or even just `run_dbt` instead of `run_dbt_and_capture` since we don't use the logs. Especially since we do use them later on, but from a different call.

Reply:
That's just a mis-copy there :) Good catch